monero/backends/jsonrpc/daemon.py

```diff
@@ -201,14 +198,1334 @@ def raw_jsonrpc_request(self, method, params=None):
+                amount = to_atomic(amount)
+            outputs[i] = {'amount': amount, 'index': index}
+
+        return self.raw_request('/get_outs', data={
+            'outputs': outputs,
+            'get_txid': get_txid})
+
+    def update(self, command, path=None):
+        """
+        Update daemon.
+
+
+        :param str command: Command to use, either "check" or "download".
+        :param str path: Optional, path where to download the update.
+
+
+        Output:
+        {
+        "auto_uri": str;
+        "hash": str;
+        "path": str; path to download the update.
+        "update": bool; States if an update is available to download (true) or not (false).
+        "user_uri": str;
+        "version": str; Version available for download.
+        "status": str; General RPC error code. "OK" means everything looks good.
+        }
+        """
+
+        command = six.ensure_text(command)
+        path = six.ensure_text(path) if path is not None else None
+
+        if command not in ('check', 'download'):
+            raise ValueError('unrecognized command: "{}"'.format(command))
+
+        data = {'command': command}
+
+        if path is not None:
+            data['path'] = path
+
+        return self.raw_request('/update', data=data)
+
+    # Supporting class methods
+
+    @classmethod
+    def get_all_known_log_categories(cls):
+        return cls._KNOWN_LOG_CATEGORIES
+
+    @classmethod
+    def get_all_known_log_levels(cls):
+        return cls._KNOWN_LOG_LEVELS
+
+    # private methods
+
+    def _set_net(self, info):
+        if info['mainnet']:
+            self._net = NET_MAIN
+        if info['testnet']:
+            self._net = NET_TEST
+        if info['stagenet']:
+            self._net = NET_STAGE
+
+    def _set_restricted(self):
+        try:
+            self.sync_info()
+
+            self._restricted = False
+        except MethodNotFound:
+            self._restricted = True
+
+    def _do_get_transactions(self, hashes, prune):
+        res = self.raw_request('/get_transactions', {
+                'txs_hashes': hashes,
+                'decode_as_json': True,
+                'prune': prune})
+        if res['status'] != 'OK':
+            raise exceptions.BackendException(res['status'])
+        txs = []
+        for tx in res.get('txs', []):
+            as_json = json.loads(tx['as_json'])
+            fee = as_json.get('rct_signatures', {}).get('txnFee')
+            txs.append(Transaction(
+                hash=tx['tx_hash'],
+                fee=from_atomic(fee) if fee else None,
+                height=None if tx['in_pool'] else tx['block_height'],
+                timestamp=datetime.fromtimestamp(
+                    tx['block_timestamp']) if 'block_timestamp' in tx else None,
+                blob=binascii.unhexlify(tx['as_hex']) or None,
+                json=as_json))
+
+        return txs
+
+    def _is_valid_256_hex(self, hexhash):
+        try:
+            bytes.fromhex(hexhash)
```

Comment:
This doesn't work with Python 2.x, where `bytes` is `str` and doesn't support `.fromhex`. I guess `bytearray` would do it.

Reply:
Okay! I will include that in the next commit.